### ISSUE TITLE
Improvement for container creation failure

### ISSUE THREAD

**Issue by jpic** [lxc/pylxd#97]
Currently, a runtime exception is raised when lxd responds with a failure to create a container. In user code, this makes it impossible to route exception handling ie.:

```
try:
    client.containers.create(dict(name='foo'))
except ContainerCreationFailure:
    print "Something went creating the container"
except:
    print "Something unknown went wrong, but the container was created"
```

Also, the runtime error doesn't carry the response, making it impossible for user code to use the HTTP response, ie. to display it:

```
try:
    client.containers.create(dict(name='foo'))
except ContainerCreationError as e:
    print "Server did not create the container, details: ", e.response.json()
```

Please let me know your thought about this, thanks again for the amazing job you're doing hosting this sprint !

**Comment by rockstar** [comment]
Yeah, I've been thinking about this a bit. Generally, I prefer built-in exceptions when I can, though `RuntimeException` is something that I'm a little concerned about overriding. I'll have to give it more thought before I give it its own special exception.

**Comment by jpic** [comment]
I think we could have more exceptions then, like ImageCreationFailed and so on. Otherwise, perhaps we could have a CreationFailed generic exception.

Anyway, what do you think of "attaching" the response to the exception ?

If you prefer attaching the response to RuntimeException I can change the PR.

**Comment by rockstar** [comment]
Fixed in #103
